internal/tfengine/tfengine.go

```diff
@@ -136,25 +141,52 @@ func dumpTemplate(conf *Config, root, outputPath string, ti *templateInfo) error
 			return err
 		}
 
-		if err := dump(rc, filepath.Dir(rp), outputPath); err != nil {
+		if err := dump(rc, filepath.Dir(rp), cacheDir, outputPath); err != nil {
 			return fmt.Errorf("recipe %q: %v", rp, err)
 		}
 
 	case ti.ComponentPath != "":
-		cp, err := pathutil.Expand(ti.ComponentPath)
+		cp, err := fetchPath(ti.ComponentPath, pwd, cacheDir)
 		if err != nil {
 			return err
 		}
-		if !filepath.IsAbs(cp) {
-			cp = filepath.Join(root, cp)
-		}
 		if err := template.WriteDir(cp, outputPath, data); err != nil {
 			return fmt.Errorf("component %q: %v", cp, err)
 		}
 	}
 	return nil
 }
 
+// fetchPath handles fetching remote paths.
+// If the path is local then it will simply expand it and return.
+// Remote paths are fetched to the cache dir.
+// Relative paths are handled from pwd.
+func fetchPath(path, pwd, cacheDir string) (string, error) {
+	if strings.HasPrefix(path, ".") { // Is local path.
```

Comment:
what if the local path is an absolute path?

Reply:
Maybe Stat the string and check if it's a file path.

Reply:
Absolute paths are not currently supported and involves some care to implement. I based the current logic on https://github.com/hashicorp/terraform/blob/807267d1b56aa6298fecf5edba98c088c2866e83/internal/initwd/getter.go#L177. Terraform does some further checks for absolute paths I think which I can port later on.

Reply:
stat seems like a good idea, I can do that

Reply:
os.Stat isn't working out too well either, I'll revisit this later.